### PR TITLE
[CUDA Graph] Deduplicate using the live executable

### PR DESCRIPTION
**Pull request by cctry**
**Problem:** Each dedup group instantiates an extra executable just to check compatibility, while grid dimensions and redundant kernel names keep compatible captures apart.

**Fix:** Validate registration against the live executable and track the graph it now represents. A rejected CUDA update preserves that executable. Leave grid validation to the update; retain kernel handles and launch attributes in the signature.

**Tested:** 17 upstream tests passed. On GB300, three grid sizes shared exactly one executable; an incompatible update was rejected, six subsequent alternating replays were correct, and cleanup destroyed the executable once. Pre-commit passed. Model-serving accuracy and performance benchmarks were not run.

<details>
<summary>Commands and actual output</summary>

With CUDA 13.0, the checkout's `python/` and `sgl-model-gateway/bindings/python/src` on `PYTHONPATH`, and `libcrypto.so.3`, `libssl.so.3`, and `libnuma.so.1` preloaded:

```bash
python -m pytest -q --disable-warnings \
  test/registered/unit/model_executor/runner_backend/test_full_cuda_graph_backend.py \
  test/registered/unit/model_executor/test_prefill_cuda_graph_runner_helpers.py
pre-commit run --files python/sglang/srt/model_executor/runner_backend/cuda_graph_dedup_mixin.py
```

```text
17 passed, 21 warnings, 2 subtests passed in 9.68s
PASS: 3 grid sizes share 1 CUDA executable; exactly 1 instantiation
PASS: incompatible update rejected; registry state preserved
PASS: 6 alternating replays remain correct after rejected update
PASS: close destroys the sole executable once and is idempotent
```

The CUDA check was run as `python B-cuda-check.py` with this script:

```python
from unittest.mock import patch

import torch

from sglang.srt.model_executor.runner_backend import cuda_graph_dedup_mixin as dedup

registry = dedup.DedupedCudaGraphRegistry()
stream = torch.cuda.Stream()
inputs = [torch.ones(n, device="cuda") for n in (1024, 8192, 32768)]
outputs = [torch.empty_like(value) for value in inputs]
bad_input = torch.ones(1024, device="cuda")
bad_output = torch.empty_like(bad_input)
torch.cuda.synchronize()
raw_graphs = []
for value, output in zip(inputs, outputs):
    graph = torch.cuda.CUDAGraph(keep_graph=True)
    with torch.cuda.graph(graph, stream=stream):
        torch.add(value, 3, out=output)
    raw_graphs.append(graph)
bad_graph = torch.cuda.CUDAGraph(keep_graph=True)
with torch.cuda.graph(bad_graph, stream=stream):
    torch.add(bad_input, 3, out=bad_output)
    bad_output.mul_(2)
torch.cuda.synchronize()
with patch.object(registry, "instantiate", wraps=registry.instantiate) as instantiate:
    graphs = [registry.register(graph) for graph in raw_graphs]
    assert registry.stats() == (3, 1), registry.stats()
    assert instantiate.call_count == 1, instantiate.call_count
print("PASS: 3 grid sizes share 1 CUDA executable; exactly 1 instantiation")
group = graphs[0].group
current = group.current_raw_graph
signature = dedup.graph_signature(raw_graphs[0].raw_cuda_graph())
with patch.object(dedup, "graph_signature", return_value=signature):
    try:
        registry.register(bad_graph)
    except AssertionError as error:
        assert "register update failed" in str(error), str(error)
    else:
        raise AssertionError("incompatible two-node graph unexpectedly accepted")
assert group.current_raw_graph == current
assert registry.stats() == (3, 1)
print("PASS: incompatible update rejected; registry state preserved")
registry.seal()
registry.seal()
for index in (2, 0, 1, 0, 2, 1):
    inputs[index].fill_(index + 7)
    graphs[index].replay()
    torch.cuda.synchronize()
    torch.testing.assert_close(outputs[index], torch.full_like(outputs[index], index + 10))
print("PASS: 6 alternating replays remain correct after rejected update")
with patch.object(registry, "destroy_exec", wraps=registry.destroy_exec) as destroy:
    registry.close()
    registry.close()
    assert destroy.call_count == 1, destroy.call_count
bad_graph.reset()
print("PASS: close destroys the sole executable once and is idempotent")
```

</details>

---
### CI States

Latest PR Test (Base): :x: [Run #34674268995](https://github.com/sgl-project/sglang/actions/runs/34674268995)
Latest PR Test (Extra): :x: [Run #34674268883](https://github.com/sgl-project/sglang/actions/runs/34674268883)
Latest PR Test (AMD ROCm 10): :x: [Run #34674269025](https://github.com/sgl-project/sglang/actions/runs/34674269025)

### CODE REVIEW

**Comment by cctry** [comment]
Duplicate of #39176, which carries the same change plus its regression test. Closing this one.
